packages/zeebe-element-templates-json-schema/src/schema.json

```diff
@@ -18,6 +18,48 @@
       "allOf": [
         {
           "$ref": "../../packages/element-templates-json-schema-shared/src/defs/base.json"
+        },
+        {
+          "if": {
+            "properties": {
+              "properties": {
+                "contains": {
+                  "properties": {
+                    "binding": {
+                      "properties": {
+                        "type": {
+                          "enum": [
+                            "bpmn:Message#property",
+                            "bpmn:Message#zeebe:subscription#property"
+                          ]
+                        }
+                      },
+                      "required": [
+                        "type"
+                      ]
+                    }
+                  },
+                  "required": [
+                    "binding"
+                  ]
+                }
+              }
+            }
+          },
+          "then": {
+            "properties": {
+              "elementType": {
+                "properties": {
+                  "eventDefinition": {
+                    "const": "bpmn:MessageEventDefinition"
+                  }
+                },
+                "required": [
+                  "eventDefinition"
+                ]
+              }
+            }
+          }
```

Comment:
This giant ensures that you put right `eventDefinition` in place once you try to add bpmn:Message bindings.

Reply:
If you have an idea how to simplify this, let me know!

Reply:
I'm going to have a little think if we can simplify this although at first glance I'm not sure. But could this be extracted to a sub-schema to not be so confusing on the main one?

Reply:
I think we do some simpler binding ifs/thens in for c7 that are `properties.json` https://github.com/camunda/element-templates-json-schema/blob/master/packages/element-templates-json-schema/src/defs/properties.json#L13

Reply:
Yeah we can extract it.

Reply:
Indeed, but here we have to define constraint on a property up the tree, i.e. the common parent is the template only :/

Reply:
OK I've extracted the schema to [template.json](packages/zeebe-element-templates-json-schema/src/defs/template.json)

Reply:
Uhm I see.

Do we need to double `properties` at the start of the `if`? I tested without it on a validator and it seemed to work
```diff
"if": {
-  "properties": {
       "properties": {
           "contains": {
```

Reply:
If I remove one of the properties, the validation does not run properly. Note that one `properties` is a JSON schema keyword while another is template's property called `properties`.

Reply:
So it reads `If property "properties" contains an item...` opposed to `If property "contains"...`

Reply:
Ah yes sorry about that. I must have tested the wrong generated schema. Makes sense. I think it's ok as is then. I'll check out the rest of the PR